src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -185,42 +200,57 @@ public void stop(Federation federation) {
 
     @PreDestroy
     public void tearDown() {
-        org.bitcoinj.core.Context.propagate(new org.bitcoinj.core.Context(ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
+        org.bitcoinj.core.Context.propagate(
+            new org.bitcoinj.core.Context(
+                ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
         peerGroup.stop();
         peerGroup = null;
     }
 
     private class BtcReleaseEthereumListener extends EthereumListenerAdapter {
         @Override
         public void onBestBlock(org.ethereum.core.Block block, List<TransactionReceipt> receipts) {
+            if (!isPegoutEnabled) {
+                return;
+            }
+
             boolean hasBetterBlockToSync = nodeBlockProcessor.hasBetterBlockToSync();
             boolean isStorageSynced = storageSynchronizer.isSynced();
             if (hasBetterBlockToSync || !isStorageSynced) {
                 logger.trace(
-                    "[onBestBlock] Node is not ready to process pegouts. hasBetterBlockToSync: {} isStorageSynced: {}",
+                    "[onBestBlock] Node is not ready to process pegouts. hasBetterBlockToSync: {} - isStorageSynced: {}",
                     hasBetterBlockToSync,
                     isStorageSynced
                 );
                 return;
             }
+            storageSynchronizer.processBlock(block, receipts);
+          
+            // Check if svp spend tx waiting for signatures is available to be signed
+            // before attempting to sign any pegouts.
+            federatorSupport.getStateForProposedFederator()
+                .map(StateForProposedFederator::getSvpSpendTxWaitingForSignatures)
+                .filter(svpSpendTxWaitingForSignatures -> isReadyToSign(block.getNumber(), svpSpendTxWaitingForSignatures.getKey()))
+                .ifPresent(svpSpendTxReadyToBeSigned -> processReleases(Set.of(svpSpendTxReadyToBeSigned)));
+
             // Processing transactions waiting for signatures on best block only still "works",
             // since it all lies within RSK's blockchain and normal rules apply. I.e., this
             // process works on a block-by-block basis.
             StateForFederator stateForFederator = federatorSupport.getStateForFederator();
-            storageSynchronizer.processBlock(block, receipts);
-
-            // Delegate processing to our own method
-            logger.trace("[onBestBlock] Got {} pegouts", stateForFederator.getRskTxsWaitingForSignatures().entrySet().size());
-            if (isPegoutEnabled) {
-                processReleases(stateForFederator.getRskTxsWaitingForSignatures().entrySet());
-            }
+            Set<Map.Entry<Keccak256, BtcTransaction>> rskTxsReadyToBeSigned = stateForFederator.getRskTxsWaitingForSignatures()
```

Comment:
```suggestion
            Set<Map.Entry<Keccak256, BtcTransaction>> pegoutsReadyToBeSigned = stateForFederator.getRskTxsWaitingForSignatures()
```

Reply:
i prefer to keep the previous name since it matches the map name

Reply:
Yes this should not only be pegouts it can also include the migration tx.